power-pay-frontend/src/hooks/StorageContext.tsx

```diff
@@ -0,0 +1,88 @@
+import { createContext, PropsWithChildren, useEffect } from 'react';
+import { useStorage } from './useStorage';
+
+interface StorageService<T> {
+  getItem: (key: string) => Promise<T | undefined>;
+  setItem: (key: string, value: T) => Promise<boolean>;
+  removeItem: (key: string) => Promise<boolean>;
+  clear: () => Promise<boolean>;
+}
+
+// Create the StorageContextData interface.
+interface StorageContextData<T> {
+  getItem: (key: string) => Promise<unknown | undefined>;
+  setItem: (key: string, value: T) => Promise<boolean>;
+  removeItem: (key: string) => Promise<boolean>;
+  clear: () => Promise<boolean>;
+}
+
+
+// Creating  a fake implementation of the StorageContextData interface.
+const fakeStorage: StorageContextData<unknown> = {
+  getItem: async (key: string): Promise< unknown | undefined> => {
+    const item = localStorage.getItem(key);
+    return item ? JSON.parse(item): undefined;
+  },
+  setItem: async (key: string, value: unknown): Promise<boolean> => {
+    localStorage.setItem(key, JSON.stringify(value));
+    return true;
+  },
+  removeItem: async (key: string): Promise<boolean> => {
+    localStorage.removeItem(key);
+    return true;
+  },
+  clear: async (): Promise<boolean> => {
+    localStorage.clear();
+    return true;
+  },
+};
+
+
+// Create the StorageContext with default functions for getItem and setItem.
+const StorageContext = createContext<StorageService<unknown> | undefined>(fakeStorage);
+
+// StorageProvider: A React component that wraps the application and provides the storage context with getItem and setItem methods.
+export function StorageProvider<T>({ initialValue, children }: PropsWithChildren<{ initialValue?: T }>) {
+  const [storedValue, setStoredValue] = useStorage<T | undefined>({ initialValue: { ...initialValue } });
+
+// Define a global constant for the key.
+  const STORAGE_KEY = 'key';
+
+  // Initializes the state with the value from the local storage if it exists.
+  useEffect(() => {
+    const item = localStorage.getItem('STORAGE_KEY');
+    if (item) {
+      setStoredValue(JSON.parse(item));
+    }
+  }, []);
```

Comment:
Here you are interacting with the localstorage. This is what I don't want, because it's hard-coded. I want to use the storage passed as props to this provider.